charts/vela-core/templates/defwithtemplate/annotations.yaml

```diff
@@ -16,17 +16,21 @@ spec:
       template: |
         // +patchStrategy=jsonMergePatch
         patch: {
-        	metadata: annotations: {
+        	let annotationsContent = {
         		for k, v in parameter {
         			(k): v
         		}
         	}
+
+        	metadata: annotations: annotationsContent
         	if context.output.spec != _|_ && context.output.spec.template != _|_ {
-        		spec: template: metadata: annotations: {
-        			for k, v in parameter {
-        				(k): v
-        			}
-        		}
+        		spec: template: metadata: annotations: annotationsContent
+        	}
+        	if context.output.spec != _|_ && context.output.spec.jobTemplate != _|_ {
```

Comment:
`&&` short-circuit is not supported in CUE, ref: https://github.com/cue-lang/cue/issues/2232
Please use `if context.output.spec != _|_ if context.output.spec.jobTemplate != _|_` instead. 
Same below.

Reply:
thanks @FogDong . Strange I didn't get an error and it worked ok 🤔 

There's also lots of traits and components that use this, not just my change! 😨 

I'll make the change though.

Reply:
PR updated